### ISSUE TITLE
[core] Make schemas more visible

### ISSUE THREAD

**Issue by romilbhardwaj** [skypilot-org/skypilot#3428]
From a user:
> If you guys rewrite the config code at some point could you use a typed system like [Pydantic settings ](https://docs.pydantic.dev/latest/concepts/pydantic_settings/)to define the structure? Makes it much easier to read the codebase
> [provided link to schemas.py] thank you! Didn't see it when skimming the code base. I have a bias for defining structure in Python code with tools that can export jsonschema like Pydantic, but it's enough just for it to be more visible.

Currently our schemas are stored in [schemas.py](https://github.com/skypilot-org/skypilot/blob/master/sky/utils/schemas.py), which is hard to find. It would be good to move schemas to something like `/sky/schemas/<schema>.json`, but using static schemas needs more thought since many fields are populated programatically.

**Comment by julien-blanchon** [comment]
Yes I agree, I would like to have a static schemas for [yaml-json-](https://github.com/redhat-developer/vscode-yaml)

**Comment by Michaelvll** [comment]
Another user requested to have skypilot task yaml schema on [jsonschema store](https://www.schemastore.org/json/). Moving to pydantic can be a way.
